internal/querycoord/replica.go

```diff
@@ -112,3 +122,102 @@ func (rep *ReplicaInfos) Remove(replicaIds ...UniqueID) {
 		}
 	}
 }
+
+// ApplyBalancePlan applies balancePlan to replica nodes.
+func (rep *ReplicaInfos) ApplyBalancePlan(p *balancePlan, kv kv.MetaKv) error {
+	rep.globalGuard.Lock()
+	defer rep.globalGuard.Unlock()
+
+	var sourceReplica, targetReplica *milvuspb.ReplicaInfo
+
+	// check source and target replica ids are valid
+	if p.sourceReplica != invalidReplicaID {
+		replica, ok := rep.replicas[p.sourceReplica]
+		if !ok {
+			return errors.New("replica not found")
+		}
+		sourceReplica = proto.Clone(replica).(*milvuspb.ReplicaInfo)
```

Comment:
This code occurs frequently, we need a private get() method just like Get().

Reply:
fixed